blog/_posts/2024-05-14-core-extension.md

```diff
@@ -0,0 +1,87 @@
+
+### Extensions vs. Add-ons
+
+Extensions and add-ons both enhance AKS functionality but differ in scope and management:
+
+- **Add-ons**: Part of the AKS resource provider (`Microsoft.ContainerService/managedClusters`), limited to AKS.
+- **Extensions**: Managed via the Extension API (`Microsoft.KubernetesConfiguration/extensions`), supporting both AKS and Arc-enabled Kubernetes environments.
+
+### What Are Core Kubernetes Extensions?
+
+ Core Kubernetes extensions are a type of Kubernetes extensions that provide broader region availability, a native AKS experience, safer version management rules, and enhanced security and efficiency on AKS. Important AKS capabilities can be supported through core Kubernetes extensions while maintaining seamless integration with AKS just like add-ons. This approach also paves the way for more services and applications to be supported across both AKS and [Arc-connected Kubernetes](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/overview) environments in the future.
+
+#### Broader region availability
+Core Kubernetes extensions are available in all regions across all clouds (Public cloud, Fairfax, Mooncake, and Air Gapped cloud). See the full list at [Azure Arc enabled Kubernetes region support](https://azure.microsoft.com/en-us/explore/global-infrastructure/products-by-region/?products=azure-arc&regions=all).
+
+#### Native AKS experience
+* Transitioning Add-ons to Core Extensions: Some AKS add-ons, such as Container Insights, Azure Managed Prometheus, and App Monitoring, are transitioning to core extensions. The user experience remains unchanged, but the underlying technology is shifting to extensions.
+
+    CLI command to enable a core Extension that transitioned from an add-on:
+    ```bash
+    az aks create/update --resource-group <group> --name <cluster> --enable-<core-extension>
+    ```
+* Shifting Regular Extensions to Core Extensions: Key extensions like Azure Backup are also being shifted to core extensions for better integration with AKS.
+
+    CLI commands:
+    * Create a regular extension:
+    ```bash
+    az k8s-extension create --extension-type <type> --resource-group <group> --cluster-name <name> --cluster-type <clusterType> --name <extension name>
+    ```
+    * Create a core extension:
+    ```bash
+    az aks extension create --extension-type <type> --resource-group <group> --cluster-name <name> --name <core extension name>
+    ```
+
+#### Safer version management rules
+Core extensions follow the same update rules as add-ons: patch versions can be upgraded within a Kubernetes minor version, while major/minor upgrades occur only with Kubernetes minor version updates to avoid introducing breaking changes in a Kubernetes version.
```

Comment:
May be worth noting that some extensions can bypass this requirement if they've demonstrated specific testing/protection against unannounced breaking changes